EtlPipelineBlobTrigger/course_stats.py

```diff
@@ -730,72 +767,70 @@ def need_unavailable(self, xml_elem):
             return True
         return False
 
-    def get_unavailable_reason_subj(self, sbj_key):
-        if sbj_key:
-            return self.get_english_sbj_label(sbj_key)
-        return "this subject"
+    def get_aggs_for_code(self, unavail_reason_code):
+        return self.unavail_reason_english["data"][unavail_reason_code].keys()
+
+    def get_unavailable(self, elem):
+        unavailable = {}
+        subj_key = elem.get(self.xml_subj_key)
+        agg = elem[self.xml_agg_key] if self.has_data(elem) else None
+        unavail_reason_code = elem[self.xml_unavail_reason_key]
+        validate_unavailable_reason_code(unavail_reason_code)
+
+        unavailable["code"] = int(unavail_reason_code)
+        unavailable["reason_english"] = self.get_unavailable_reason_str(
+            unavail_reason_code, subj_key, agg, elem
+        )
+        unavailable["reason_welsh"] = self.get_unavailable_reason_str(
+            unavail_reason_code, subj_key, agg, elem, welsh=True
+        )
+        return unavailable
 
     def get_unavailable_reason_str(
-        self, unavail_reason_code, subj_key, agg, xml_elem
+        self, unavail_reason_code, subj_key, agg, xml_elem, welsh=False
```

Comment:
If im understanding this correctly, should welsh really be hardcoded to False? Or should the logic actually be add the welsh unavailability reason even if welsh lookup is missing a reason (e.g. empty string)?

Reply:
This is using Python's default argument values. If the caller doesn't specify a value for welsh it defaults to False so the english lookup will be used.